public/components/FeatureAnywhereContextMenu/AssociatedDetectors/containers/AssociatedDetectors.tsx

```diff
@@ -0,0 +1,337 @@
+        const savedAugmentForThisVisualization: ISavedAugmentVis[] =
+          savedAugmentObjects.filter(
+            (savedObj) => get(savedObj, 'visId', '') === embeddable.vis.id
+          );
+
+        // find saved Augment object matching detector we want to unlink
+        // There should only be one detector and vis pairing
+        const savedAugmentToUnlink = savedAugmentForThisVisualization.find(
+          (savedObject) =>
+            get(savedObject, 'pluginResourceId', '') === detectorToUnlink.id
+        );
+        const savedObjectToUnlinkId = get(savedAugmentToUnlink, 'id', '');
+        await savedObjectLoader
+          .delete(savedObjectToUnlinkId)
+          .then(async (resp: any) => {
+            core.notifications.toasts.addSuccess({
+              title: `Association removed between the ${detectorToUnlink.name}
+              and the ${embeddableTitle} visualization`,
+              text: "The detector's anomalies do not appear on the visualization. Refresh your dashboard to update the visualization",
+            });
+          })
+          .catch((error) => {
+            core.notifications.toasts.addDanger(
+              prettifyErrorMessage(
+                `Error unlinking selected detector: ${error}`
+              )
+            );
+          })
+          .finally(() => {
+            getDetectors();
+          });
+      }
+    });
+  };
+
+  const handleHideModal = () => {
+    setConfirmModalState({
+      ...confirmModalState,
+      isOpen: false,
+    });
+  };
+
+  const handleConfirmModal = () => {
+    setConfirmModalState({
+      ...confirmModalState,
+      isRequestingToClose: true,
+    });
+  };
+
+  const getDetectors = async () => {
+    dispatch(getDetectorList(GET_ALL_DETECTORS_QUERY_PARAMS));
+  };
+
+  // TODO: this part is incomplete because it is pending on a different PR that will have all the associate existing changes
+  const openAssociateDetectorFlyout = async () => {
+    console.log('inside create anomaly detector');
+  };
+
+  const handleUnlinkDetectorAction = (detector: DetectorListItem) => {
+    setDetectorToUnlink(detector);
+    setConfirmModalState({
+      isOpen: true,
+      action: ASSOCIATED_DETECTOR_ACTION.UNLINK,
+      isListLoading: false,
+      isRequestingToClose: false,
+      affectedDetector: detector,
+    });
+  };
+
+  const columns = useMemo(
+    () => getColumns({ handleUnlinkDetectorAction }),
+    [handleUnlinkDetectorAction]
+  );
+
+  const renderEmptyMessage = () => {
+    if (isLoading) {
+      return 'Loading detectors...';
+    } else if (!isEmpty(selectedDetectors)) {
+      return (
+        <EmptyAssociatedDetectorMessage
+          isFilterApplied={true}
+          embeddableTitle={embeddableTitle}
+        />
+      );
+    } else {
+      return (
+        <EmptyAssociatedDetectorMessage
+          isFilterApplied={false}
+          embeddableTitle={embeddableTitle}
+        />
+      );
+    }
```

Comment:
This is to catch if there is anything in the search box, if we are rendering the empty message we need to know if its because there is no selected detectors in the state at all or if its because the user's search result is returning nothing. The message would then differ with something like "There are no detectors matching your search" vs "There are no anomaly detectors associated with ${props.embeddableTitle} visualization."

Reply:
The renderEmptyMessage() is automatically called by the EuiInMemoryTable when there is no data to display in table

Reply:
Got it, that makes sense